### PR TITLE
Reorganize `game` module

### PR DESCRIPTION
**Pull request by janhohenheim**
Part of https://github.com/TheBevyFlock/bevy_quickstart/issues/203
This makes the `README.md` outdated. I'll fix that in a follow-up.

### CODE REVIEW

File: src/demo/mod.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//! Demo gameplay. All of these modules are intented do be here only for demonstration purposes
```
Comment (benfrankel) [inline src/demo/mod.rs:1]:
```suggestion
//! Demo gameplay. All of these modules are only intended for demonstration purposes
```

File: src/demo/movement.rs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 //!
 //! In our case, the character controller has the following logic:
 //! - Set [`MovementController`] intent based on directional keyboard input.
+//!     This is done in the `player` module, as it is specific to the player character.
```
Comment (benfrankel) [inline src/demo/movement.rs:6]: Well, unless we add `With<Player>` to the query, _technically_ it's not specific to the player character rn.

File: src/screens/playing.rs [hunk 3]
```diff
@@ -3,9 +3,8 @@
 use bevy::{input::common_conditions::input_just_pressed, prelude::*};
 
 use super::Screen;
-use crate::game::{
-    assets::SoundtrackHandles, audio::soundtrack::SoundtrackCommands as _, spawn::level::SpawnLevel,
-};
+use crate::demo::level::SpawnLevel;
+use crate::{assets::SoundtrackHandles, audio::soundtrack::SoundtrackCommands as _};
```
Comment (benfrankel) [inline src/screens/playing.rs:7]: I like 1 line per import personally, but should merge these to stay consistent

File: src/ui/widgets.rs [hunk 4]
```diff
@@ -6,7 +6,7 @@ use bevy::{
     ui::Val::*,
 };
 
-use super::{interaction::InteractionPalette, palette::*};
+use super::{interaction::InteractionPalette, palette::*, prelude::OnPress};
```
Comment (benfrankel) [inline src/ui/widgets.rs:9]:
```suggestion
use super::{interaction::{InteractionPalette, OnPress}, palette::*};
```

**Review by benfrankel — CHANGES_REQUESTED** [review]

File: src/screens/playing.rs (hunk 3 above)
Comment (janhohenheim) [reply src/screens/playing.rs]: Ran nightly cargo fmt.

File: src/ui/widgets.rs (hunk 4 above)
Comment (janhohenheim) [reply src/ui/widgets.rs:9]: Ran nightly cargo fmt.

File: src/demo/movement.rs (hunk 2 above)
Comment (janhohenheim) [reply src/demo/movement.rs]: [`ff2dbb3` (#236)](https://github.com/TheBevyFlock/bevy_quickstart/pull/236/commits/ff2dbb31f444d407c9a6f9442b0e91f7adec2f4f)

File: src/ui/widgets.rs (hunk 4 above)
Comment (benfrankel) [reply src/ui/widgets.rs:9]: This isn't resolved by `fmt`, would need to change the import to be from `interaction` instead of `prelude` first.

Comment (janhohenheim) [reply src/ui/widgets.rs:9]: Oh, right. Thanks.

**Review by benfrankel — APPROVED** [review]

**Pull request merged**